client/store/boat.js

```diff
@@ -30,20 +32,41 @@ export const fetchBoats = () => async dispatch => {
   }
 };
 
-export const getSingleBoat = id => async dispatch => {
-  try {
-    const { data } = await axios.get(`/api/boats/${id}`);
-    dispatch(gotSingleBoat(data));
-  } catch (error) {
-    console.error(error);
-  }
+export const getSingleBoat = id => {
+  return async dispatch => {
+    try {
+      const { data } = await axios.get(`/api/boats/${id}`);
+      dispatch(gotSingleBoat(data));
+    } catch (error) {
+      console.error(error);
+    }
+  };
 };
 
-export default function(state = initialState, action) {
+/**
+ * REDUCER
+ */
+function boatsReducer(boats = [], action) {
   switch (action.type) {
     case GET_BOATS:
       return action.boats;
     default:
-      return state;
+      return boats;
```

Comment:
@amantina05 did you fix josh's reducer?

Reply:
oh wait nvm you are using combine reducers

Reply:
@amantina05 we want immutable copies... so instead of returning boats maybe
[...boats] ?